modules/general/storage-account-rbac/main.bicep

```diff
@@ -30,18 +30,13 @@ param roleAssignmentId string = ''
 targetScope = 'resourceGroup'
 
 
-// provide a more user-friendly roleDefinitionId lookup
-var roles = {
-  'Owner': resourceId('Microsoft.Authorization/roleDefinitions', '8e3af657-a8ff-443c-a75c-2fe8c4bcb635')
-  'Reader': resourceId('Microsoft.Authorization/roleDefinitions', 'acdd72a7-3385-48ef-bd42-f606fba81ae7')
-  'Contributor': resourceId('Microsoft.Authorization/roleDefinitions', 'b24988ac-6180-42a0-ab88-20f7382dd24c')
-  'Storage Blob Data Owner': resourceId('Microsoft.Authorization/roleDefinitions', 'b7e6dc6d-f1e8-4753-8033-0f276bb0955b')
-  'Storage Blob Data Contributor': resourceId('Microsoft.Authorization/roleDefinitions', 'ba92f5b4-2d11-453d-a403-e96b0029c9fe')
-  'Storage Blob Data Reader': resourceId('Microsoft.Authorization/roleDefinitions', '2a2b9908-6ea1-4ae2-8e65-a410df84e7d1')
+module role_definitions '../rbac-built-in-roles/main.bicep' = {
+  name: 'roleDefinitions'
 }
+
 // generate a stable Id for a given role assignment
 var storageAccountId = resourceId('Microsoft.Storage/storageAccounts', storageAccountName)
-var generatedRoleAssignmentId = guid('${storageAccountId}${roles[role]}${assigneeObjectId}')
+var generatedRoleAssignmentId = guid('${storageAccountId}${role}${assigneeObjectId}')
```

Comment:
I wonder if the `guid` function is case sensitive?  Will it create a different GUID if the role name was specified with inconsistent casing?

Reply:
Interesting - we shouldn't be able to get into a situation where this matters because the Bicep parameter validation is case sensitive (and we are validating the allowed values for `role`). However, ARM itself isn't case sensitive, so I guess you could potentially get into trouble if you bypassed Bicep.

I'll add a `toLower` function just to be safe.